test_hashing fails randomly

The test_hasing pytest seems to fail randomly. It can be run in isolation and works, but fails when run as part of the complete test suite.

```
15:16:47 =================================== FAILURES ===================================
15:16:47 _________________________________ test_hashing _________________________________
15:16:47 
15:16:47     def test_hashing():
15:16:47         # Make data
15:16:47         nrows = 8
15:16:47         dtypes = [np.int8, np.int16, np.int32, np.int64, np.float32]#, np.float64] # dtypes as number of columns
15:16:47     
15:16:47         # pytest for hashing fails if called collectively on folder and if number of columns>5, but works well if called seperately like
15:16:47         # pytest --cache-clear -vvs ~/libgdf/python/tests/test_hashing.py
15:16:47     
15:16:47         hash_input = []
15:16:47         for dt in dtypes:
15:16:47             hi = np.array(np.random.randint(0, 28, nrows), dtype=dt)
15:16:47             hi[-1] = hi[0]
15:16:47             hash_input.append(hi)
15:16:47     
15:16:47         ncols = len(hash_input)
15:16:47         magic = libgdf.GDF_HASH_MURMUR3
15:16:47     
15:16:47         with _make_hash_input(hash_input, ncols) as col_input:
15:16:47             # Hash
15:16:47             hashed_column = _call_hash_multi(libgdf.gdf_hash, ncols, col_input, magic, nrows)
15:16:47     
15:16:47         # Check if first and last row are equal
15:16:47 >       assert tuple([hashed_column[0]]) == tuple([hashed_column[-1]])
15:16:47 E       assert (-479244108,) == (-1102096448,)
15:16:47 E         At index 0 diff: -479244108 != -1102096448
15:16:47 E         Full diff:
15:16:47 E         - (-479244108,)
15:16:47 E         + (-1102096448,)
15:16:47 
15:16:47 test_hashing.py:95: AssertionError
```

--- a/python/tests/test_hashing.py
+++ b/python/tests/test_hashing.py
@@ -76,7 +76,6 @@ def test_hashing():
     nrows = 8
     dtypes = [np.int8, np.int16, np.int32, np.int64, np.float32, np.float64]  # dtypes as number of columns
 
-    np.random.seed(None)
     hash_input = []
     for dt in dtypes:
         if(dt == np.float32):
@@ -87,10 +86,6 @@ def test_hashing():
             hi = np.array(np.random.randint(np.iinfo(dt).min, np.iinfo(dt).max, nrows), dtype=dt)
         hi[-1] = hi[0]
         hash_input.append(hi)
-
-    print("True Input:")
-    for i in range(len(hash_input)):
-        print(hash_input[i])
 
     ncols = len(hash_input)
     magic = libgdf.GDF_HASH_MURMUR3

--- a/python/tests/test_hashing.py
+++ b/python/tests/test_hashing.py
@@ -18,8 +18,6 @@ from __future__ import print_function
 import ctypes
 from contextlib import contextmanager
 
-import pytest
-
 import numpy as np
 from numba import cuda
 
@@ -31,18 +29,21 @@ from .utils import new_column, unwrap_devary, get_dtype
 @contextmanager
 def _make_hash_input(hash_input, ncols):
     ci = []
+    di = []
+    for i in range(ncols):
+        di.append(cuda.to_device(hash_input[i]))
 
     for i in range(ncols):
-        d_input = cuda.to_device(hash_input[i])
         col_input = new_column()
-        libgdf.gdf_column_view(col_input, unwrap_devary(d_input), ffi.NULL,
-                            hash_input[i].size, get_dtype(d_input.dtype))
+        libgdf.gdf_column_view(col_input, unwrap_devary(di[i]), ffi.NULL,
+                               hash_input[i].size, get_dtype(hash_input[i].dtype))
         ci.append(col_input)
 
     yield ci
 
+
 def _call_hash_multi(api, ncols, col_input, magic, nrows):
-    out_ary = np.zeros(nrows,dtype=np.int32)
+    out_ary = np.zeros(nrows, dtype=np.int32)
     d_out = cuda.to_device(out_ary)
     col_out = new_column()
     libgdf.gdf_column_view(col_out, unwrap_devary(d_out), ffi.NULL,
@@ -69,28 +70,34 @@ def _call_hash_multi(api, ncols, col_input, magic, nrows):
 
     return hashed_result
 
-@pytest.mark.skip(reason='Fails randomly, see JIRA issue RAPETL-47')
+
 def test_hashing():
     # Make data
     nrows = 8
-    dtypes = [np.int8, np.int16, np.int32, np.int64, np.float32]#, np.float64] # dtypes as number of columns
+    dtypes = [np.int8, np.int16, np.int32, np.int64, np.float32, np.float64]  # dtypes as number of columns
 
-    # pytest for hashing fails if called collectively on folder and if number of columns>5, but works well if called seperately like
-    # pytest --cache-clear -vvs ~/libgdf/python/tests/test_hashing.py
-
+    np.random.seed(None)
     hash_input = []
     for dt in dtypes:
-        hi = np.array(np.random.randint(0, 28, nrows), dtype=dt)
+        if(dt == np.float32):
+            hi = np.array(np.random.randint(np.iinfo(np.int32).min, np.iinfo(np.int32).max, nrows), dtype=dt)
+        elif(dt == np.float64):
+            hi = np.array(np.random.randint(np.iinfo(np.int64).min, np.iinfo(np.int64).max, nrows), dtype=dt)
+        else:
+            hi = np.array(np.random.randint(np.iinfo(dt).min, np.iinfo(dt).max, nrows), dtype=dt)
         hi[-1] = hi[0]
         hash_input.append(hi)
 
+    print("True Input:")
+    for i in range(len(hash_input)):
+        print(hash_input[i])
+
     ncols = len(hash_input)
     magic = libgdf.GDF_HASH_MURMUR3
-    
+
     with _make_hash_input(hash_input, ncols) as col_input:
         # Hash
         hashed_column = _call_hash_multi(libgdf.gdf_hash, ncols, col_input, magic, nrows)
 
     # Check if first and last row are equal
     assert tuple([hashed_column[0]]) == tuple([hashed_column[-1]])
-    

